In [3]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

### Let`s review each df one by one and summarize key insights

## Customers dataset

In [62]:
# Load the datasets
# Customers dataset
customers_df = pd.read_csv('/Users/irasafonik/SMART_business/customers.csv')
customers_df.head()

customer_id                customer_unique_id  \
0  06b8999e2fba1a1fbc88172c00ba8bc7  861eff4711a542e4b93843c6dd7febb0   
1  18955e83d337fd6b2def6b18a428ac77  290c77bc529b7ac935b93aa66c333dc3   
2  4e7b3e00288586ebd08712fdd0374a03  060e732b5b29e8181a18229c7b0b2b5e   
3  b2b6027bc5c5109e529d4dc6358b12c3  259dac757896d24d7702b9acbbff3f3c   
4  4f2d8ab171c80ec8364f7c12e35b23ad  345ecd01c38d18a9036ed96c73b8d066   

   customer_zip_code_prefix          customer_city customer_state  
0                     14409                 franca             SP  
1                      9790  sao bernardo do campo             SP  
2                      1151              sao paulo             SP  
3                      8775        mogi das cruzes             SP  
4                     13056               campinas             SP

In [64]:
# Function to display basic information about a dataframe
customers_df.info()
customers_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 5 columns):
 #   Column                    Non-Null Count  Dtype 
---  ------                    --------------  ----- 
 0   customer_id               99441 non-null  object
 1   customer_unique_id        99441 non-null  object
 2   customer_zip_code_prefix  99441 non-null  int64 
 3   customer_city             99441 non-null  object
 4   customer_state            99441 non-null  object
dtypes: int64(1), object(4)
memory usage: 3.8+ MB


customer_id                 0
customer_unique_id          0
customer_zip_code_prefix    0
customer_city               0
customer_state              0
dtype: int64

In [65]:
customers_df.duplicated().sum()

0

## 📝 Conclusion

### Customers Dataset Overview

- The dataset contains **99,441 entries** with **5 columns**:
  - `customer_id` - key to the orders dataset. Each order has a unique customer_id
  - `customer_unique_id` - unique identifier of a customer
  - `customer_zip_code_prefix` - first five digits of customer zip code
  - `customer_city` - customer city name
  - `customer_state` - customer state

### Data Quality

- The dataset is clean with **no missing values**.
- There are **no duplicate rows** in the dataset.

### Insights

- The data provides a comprehensive view of customer demographics, with a focus on geographical distribution.

### Next Steps

- Conduct further analysis on purchasing patterns.
- Investigate customer behavior.
- Explore sales trends.

## Geolocation dataset

In [17]:
# Geolocation dataset
geolocation_df = pd.read_csv('/Users/irasafonik/SMART_business/geolocation.csv')
geolocation_df.head()

geolocation_zip_code_prefix  geolocation_lat  geolocation_lng  \
0                         1037       -23.545621       -46.639292   
1                         1046       -23.546081       -46.644820   
2                         1046       -23.546129       -46.642951   
3                         1041       -23.544392       -46.639499   
4                         1035       -23.541578       -46.641607   

  geolocation_city geolocation_state  
0        sao paulo                SP  
1        sao paulo                SP  
2        sao paulo                SP  
3        sao paulo                SP  
4        sao paulo                SP

In [67]:
geolocation_df.info()
geolocation_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1000163 entries, 0 to 1000162
Data columns (total 5 columns):
 #   Column                       Non-Null Count    Dtype  
---  ------                       --------------    -----  
 0   geolocation_zip_code_prefix  1000163 non-null  int64  
 1   geolocation_lat              1000163 non-null  float64
 2   geolocation_lng              1000163 non-null  float64
 3   geolocation_city             1000163 non-null  object 
 4   geolocation_state            1000163 non-null  object 
dtypes: float64(2), int64(1), object(2)
memory usage: 38.2+ MB


geolocation_zip_code_prefix    0
geolocation_lat                0
geolocation_lng                0
geolocation_city               0
geolocation_state              0
dtype: int64

In [68]:
geolocation_df.duplicated().sum()

261831

## 📝 Conclusion

### Geolocation Dataset Overview

- The dataset contains **1,000,163 entries** with **5 columns**:
    - `geolocation_zip_code_prefix`: First 5 digits of zip code
    - `geolocation_lat`: Latitude
    - `geolocation_lng`: Longitude
    - `geolocation_city`: City name
    - `geolocation_state`: State

### Data Quality

- There are **no missing values** in this dataset.
- There are **261,831 duplicate rows** in the dataset.

### Insights

- This dataset provides precise geolocation data that can be used for spatial analysis.

### Next Steps

- Duplicate rows should be analysed.
- Use this geolocation data to map customer and seller locations.
- Analyze the geographic distribution of orders.

## Order items dataset

In [96]:
# Order items dataset
order_items_df = pd.read_csv('/Users/irasafonik/SMART_business/order_items.csv')
order_items_df.head()

order_id  order_item_id  \
0  00010242fe8c5a6d1ba2dd792cb16214              1   
1  00018f77f2f0320c557190d7a144bdd3              1   
2  000229ec398224ef6ca0657da4fc703e              1   
3  00024acbcdf0a6daa1e931b038114c75              1   
4  00042b26cf59d7ce69dfabb4e55b4fd9              1   

                         product_id                         seller_id  \
0  4244733e06e7ecb4970a6e2683c13e61  48436dade18ac8b2bce089ec2a041202   
1  e5f2d52b802189ee658865ca93d83a8f  dd7ddc04e1b6c2c614352b383efe2d36   
2  c777355d18b72b67abbeef9df44fd0fd  5b51032eddd242adc84c38acab88f23d   
3  7634da152a4610f1595efa32f14722fc  9d7a1d34a5052409006425275ba1c2b4   
4  ac6c3623068f30de03045865e4e10089  df560393f3a51e74553ab94004ba5c87   

   shipping_limit_date   price  freight_value  
0  2017-09-19 09:45:35   58.90          13.29  
1  2017-05-03 11:05:13  239.90          19.93  
2  2018-01-18 14:48:30  199.00          17.87  
3  2018-08-15 10:10:18   12.99          12.79  
4  2017-02-13 13:57:51  199.90          18.14

In [28]:
order_items_df.info()
order_items_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 112650 entries, 0 to 112649
Data columns (total 7 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   order_id             112650 non-null  object 
 1   order_item_id        112650 non-null  int64  
 2   product_id           112650 non-null  object 
 3   seller_id            112650 non-null  object 
 4   shipping_limit_date  112650 non-null  object 
 5   price                112650 non-null  float64
 6   freight_value        112650 non-null  float64
dtypes: float64(2), int64(1), object(4)
memory usage: 6.0+ MB


order_id               0
order_item_id          0
product_id             0
seller_id              0
shipping_limit_date    0
price                  0
freight_value          0
dtype: int64

In [69]:
order_items_df.duplicated().sum()

0

## 📝 Conclusion

### Order Items Dataset Overview

- The dataset contains **112,650 entries** with **7 columns**:
    - `order_id`: Order unique identifier
    - `order_item_id`: Sequential number identifying the number of items included in the same order
    - `product_id`: Product unique identifier
    - `seller_id`: Seller unique identifier
    - `shipping_limit_date`: Shows the seller shipping limit date for handling the order over to the logistic partner
    - `price`: Item price
    - `freight_value`: Item freight value (if an order has more than one item the freight value is split between items)

### Data Quality

- There are **no missing values** in this dataset.
- There are **no duplicate rows** in the dataset.

### Insights

- The dataset provides detailed information about individual items within each order.
- Each row represents an item in an order, with details on pricing and freight value.
- The `shipping_limit_date` indicates the latest date by which the seller must hand over the item to the logistics partner.

### Next Steps

- **Formatting to Datetime Type:** `shipping_limit_date` need formatting to datetime type, currently of type 'object'.
- Analyze item-level data to understand product-level sales and pricing patterns.
- Investigate the relationship between freight values and shipping limits.
- Use the `order_id` to join with other datasets for comprehensive order analysis.


## Order payments dataset

In [29]:
# Order payments dataset
order_payments_df = pd.read_csv('/Users/irasafonik/SMART_business/order_payments.csv')
order_payments_df.head()

order_id  payment_sequential payment_type  \
0  b81ef226f3fe1789b1e8b2acac839d17                   1  credit_card   
1  a9810da82917af2d9aefd1278f1dcfa0                   1  credit_card   
2  25e8ea4e93396b6fa0d3dd708e76c1bd                   1  credit_card   
3  ba78997921bbcdc1373bb41e913ab953                   1  credit_card   
4  42fdf880ba16b47b59251dd489d4441a                   1  credit_card   

   payment_installments  payment_value  
0                     8          99.33  
1                     1          24.39  
2                     1          65.71  
3                     8         107.78  
4                     2         128.45

In [30]:
order_payments_df.info()
order_payments_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 103886 entries, 0 to 103885
Data columns (total 5 columns):
 #   Column                Non-Null Count   Dtype  
---  ------                --------------   -----  
 0   order_id              103886 non-null  object 
 1   payment_sequential    103886 non-null  int64  
 2   payment_type          103886 non-null  object 
 3   payment_installments  103886 non-null  int64  
 4   payment_value         103886 non-null  float64
dtypes: float64(1), int64(2), object(2)
memory usage: 4.0+ MB


order_id                0
payment_sequential      0
payment_type            0
payment_installments    0
payment_value           0
dtype: int64

In [70]:
order_payments_df.duplicated().sum()

0

## 📝 Conclusion

### Order Payments Dataset Overview

- The dataset contains **103,886 entries** with **5 columns**:
    - `order_id`: Unique identifier of an order
    - `payment_sequential`: A customer may pay an order with more than one payment method. If they do so, a sequence will be created
    - `payment_type`: Method of payment chosen by the customer
    - `payment_installments`: Number of installments chosen by the customer
    - `payment_value`: Transaction value

### Data Quality

- There are **no missing values** in this dataset.
- There are **no duplicate rows** in the dataset.

### Insights

- This dataset provides detailed information on payment transactions for each order.
- Each row represents a payment transaction, including the method and value of the payment.
- The `payment_sequential` column helps track multiple payment methods used for a single order.

### Next Steps

- Analyze the distribution of payment types and their frequencies.
- Investigate the impact of payment installments on the total payment value.
- Use the `order_id` to join with other datasets for comprehensive order and payment analysis.


## Order reviews dataset

In [57]:
# Order reviews dataset
order_reviews_df = pd.read_csv('/Users/irasafonik/SMART_business/order_reviews.csv')
order_reviews_df.head()

review_id                          order_id  \
0  7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1  80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2  228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3  e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4  f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   

   review_score review_comment_title  \
0             4                  NaN   
1             5                  NaN   
2             5                  NaN   
3             5                  NaN   
4             5                  NaN   

                              review_comment_message review_creation_date  \
0                                                NaN  2018-01-18 00:00:00   
1                                                NaN  2018-03-10 00:00:00   
2                                                NaN  2018-02-17 00:00:00   
3              Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4  Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   

  review_answer_timestamp  
0     2018-01-18 21:46:59  
1     2018-03-11 03:05:13  
2     2018-02-18 14:36:24  
3     2017-04-21 22:02:06  
4     2018-03-02 10:26:53

In [32]:
order_reviews_df.info()
order_reviews_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 100000 entries, 0 to 99999
Data columns (total 7 columns):
 #   Column                   Non-Null Count   Dtype 
---  ------                   --------------   ----- 
 0   review_id                100000 non-null  object
 1   order_id                 100000 non-null  object
 2   review_score             100000 non-null  int64 
 3   review_comment_title     11715 non-null   object
 4   review_comment_message   41753 non-null   object
 5   review_creation_date     100000 non-null  object
 6   review_answer_timestamp  100000 non-null  object
dtypes: int64(1), object(6)
memory usage: 5.3+ MB


review_id                      0
order_id                       0
review_score                   0
review_comment_title       88285
review_comment_message     58247
review_creation_date           0
review_answer_timestamp        0
dtype: int64

In [71]:
order_reviews_df.duplicated().sum()

0

## 📝 Conclusion

### Dataset Overview

- The dataset contains **100,000 entries** with **7 columns**:
  - `review_id` - unique review identifier
  - `order_id`- unique order identifier
  - `review_score` - Note ranging from 1 to 5 given by the customer on a satisfaction survey.
  - `review_comment_title` - Comment title from the review left by the customer, in Portuguese.
  - `review_comment_message` - Comment message from the review left by the customer, in Portuguese.
  - `review_creation_date` - Shows the date in which the satisfaction survey was sent to the customer.
  - `review_answer_timestamp` - Shows satisfaction survey answer timestamp.

### Data Quality

- There are **no duplicate rows** in the dataset.
- **Significant missing values** are present in `review_comment_title` and `review_comment_message`, with many entries lacking review comments.

#### Analysis of `order_reviews_df`:

- **`review_comment_title`:**
  - 88% of records have no review title (88,285 missing values out of 100,000 records).
  - Only 11,715 non-null values.

- **`review_comment_message`:**
  - 58% of records have no review message (58,247 missing values out of 100,000 records).
  - 41,753 non-null values.

- **`review_score`** is complete, indicating that all reviews have scores.

### Next Steps

#### Handling Missing Values:
- Fill missing values for review titles with `"No Title"`.
- Fill missing values for review messages with `"No Comment"`.

#### Analyze Comment Lengths:
- For non-null values, analyze the length of titles and messages to understand how detailed the reviews are.

#### Text Analysis:
- For non-null comments, consider performing sentiment analysis or keyword extraction.

#### Correlation with Review Score:
- Investigate if there's a relationship between the presence of a comment and the review score.

#### Formatting to Datetime Type:
- `review_creation_date` and `review_answer_timestamp` need formatting to datetime type. Both columns are currently of type 'object'.


## Orders dataset

In [33]:
# Orders dataset
orders_df = pd.read_csv('/Users/irasafonik/SMART_business/orders.csv')
orders_df.head()

order_id                       customer_id  \
0  e481f51cbdc54678b7cc49136f2d6af7  9ef432eb6251297304e76186b10a928d   
1  53cdb2fc8bc7dce0b6741e2150273451  b0830fb4747a6c6d20dea0b8c802d7ef   
2  47770eb9100c2d0c44946d9cf07ec65d  41ce2a54c0b03bf3443c3d931a367089   
3  949d5b44dbf5de918fe9c16f97b45f8a  f88197465ea7920adcdbec7375364d82   
4  ad21c59c0840e6cb83a9ceb5573f8159  8ab97904e6daea8866dbdbc4fb7aad2c   

  order_status order_purchase_timestamp    order_approved_at  \
0    delivered      2017-10-02 10:56:33  2017-10-02 11:07:15   
1    delivered      2018-07-24 20:41:37  2018-07-26 03:24:27   
2    delivered      2018-08-08 08:38:49  2018-08-08 08:55:23   
3    delivered      2017-11-18 19:28:06  2017-11-18 19:45:59   
4    delivered      2018-02-13 21:18:39  2018-02-13 22:20:29   

  order_delivered_carrier_date order_delivered_customer_date  \
0          2017-10-04 19:55:00           2017-10-10 21:25:13   
1          2018-07-26 14:31:00           2018-08-07 15:27:45   
2          2018-08-08 13:50:00           2018-08-17 18:06:29   
3          2017-11-22 13:39:59           2017-12-02 00:28:42   
4          2018-02-14 19:46:34           2018-02-16 18:17:02   

  order_estimated_delivery_date  
0           2017-10-18 00:00:00  
1           2018-08-13 00:00:00  
2           2018-09-04 00:00:00  
3           2017-12-15 00:00:00  
4           2018-02-26 00:00:00

In [34]:
orders_df.info()
orders_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99441 entries, 0 to 99440
Data columns (total 8 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   order_id                       99441 non-null  object
 1   customer_id                    99441 non-null  object
 2   order_status                   99441 non-null  object
 3   order_purchase_timestamp       99441 non-null  object
 4   order_approved_at              99281 non-null  object
 5   order_delivered_carrier_date   97658 non-null  object
 6   order_delivered_customer_date  96476 non-null  object
 7   order_estimated_delivery_date  99441 non-null  object
dtypes: object(8)
memory usage: 6.1+ MB


order_id                            0
customer_id                         0
order_status                        0
order_purchase_timestamp            0
order_approved_at                 160
order_delivered_carrier_date     1783
order_delivered_customer_date    2965
order_estimated_delivery_date       0
dtype: int64

In [72]:
orders_df.duplicated().sum()

0

## 📝 Conclusion

### Orders Dataset Overview

- The dataset contains **99,441 entries** with **8 columns**:
    - `order_id`: Unique identifier of the order
    - `customer_id`: Key to the customer dataset; each order has a unique `customer_id`
    - `order_status`: Reference to the order status (delivered, shipped, etc.)
    - `order_purchase_timestamp`: Timestamp when the purchase was made
    - `order_approved_at`: Timestamp when the payment was approved
    - `order_delivered_carrier_date`: Timestamp when the order was handed over to the logistics partner
    - `order_delivered_customer_date`: Actual date the order was delivered to the customer
    - `order_estimated_delivery_date`: Estimated delivery date provided to the customer at the time of purchase

### Data Quality

- There are **no duplicate rows** in this dataset.
- **Missing values** are present in the following columns:
    - `order_approved_at`
    - `order_delivered_carrier_date`
    - `order_delivered_customer_date`

  These missing values may impact analysis related to delivery performance and order processing.

### Next Steps

- **Formatting to Datetime Type:** `order_delivered_carrier_date` and `order_delivered_customer_date` and `order_estimated_delivery_date` need to be formatting to datetime type, currently of type 'object'.
- Address missing values in delivery-related columns to ensure completeness of delivery analysis.
- Analyze the distribution of `order_status` and `order_purchase_timestamp` to understand order processing times.
- Investigate delivery delays by comparing `order_delivered_customer_date` with `order_estimated_delivery_date` and identify factors affecting timely delivery.


## Products dataset

In [35]:
# Products dataset
products_df = pd.read_csv('/Users/irasafonik/SMART_business/products.csv')
products_df.head()

product_id  product_category_name  \
0  1e9e8ef04dbcff4541ed26657ea517e5             perfumaria   
1  3aa071139cb16b67ca9e5dea641aaa2f                  artes   
2  96bd76ec8810374ed1b65e291975717f          esporte_lazer   
3  cef67bcfe19066a932b7673e239eb23d                  bebes   
4  9dc1a7de274444849c219cff195d0b71  utilidades_domesticas   

   product_name_lenght  product_description_lenght  product_photos_qty  \
0                 40.0                       287.0                 1.0   
1                 44.0                       276.0                 1.0   
2                 46.0                       250.0                 1.0   
3                 27.0                       261.0                 1.0   
4                 37.0                       402.0                 4.0   

   product_weight_g  product_length_cm  product_height_cm  product_width_cm  
0             225.0               16.0               10.0              14.0  
1            1000.0               30.0               18.0              20.0  
2             154.0               18.0                9.0              15.0  
3             371.0               26.0                4.0              26.0  
4             625.0               20.0               17.0              13.0

In [36]:
products_df.info()
products_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 32951 entries, 0 to 32950
Data columns (total 9 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   product_id                  32951 non-null  object 
 1   product_category_name       32341 non-null  object 
 2   product_name_lenght         32341 non-null  float64
 3   product_description_lenght  32341 non-null  float64
 4   product_photos_qty          32341 non-null  float64
 5   product_weight_g            32949 non-null  float64
 6   product_length_cm           32949 non-null  float64
 7   product_height_cm           32949 non-null  float64
 8   product_width_cm            32949 non-null  float64
dtypes: float64(7), object(2)
memory usage: 2.3+ MB


product_id                      0
product_category_name         610
product_name_lenght           610
product_description_lenght    610
product_photos_qty            610
product_weight_g                2
product_length_cm               2
product_height_cm               2
product_width_cm                2
dtype: int64

In [73]:
products_df.duplicated().sum()

0

## 📝 Conclusion

### Products Dataset Overview

- The dataset contains **32,951 entries** with **9 columns**:
    - `product_id`: Unique product identifier
    - `product_category_name`: Root category of the product (in Portuguese)
    - `product_name_lenght`: Number of characters in the product name
    - `product_description_lenght`: Number of characters in the product description
    - `product_photos_qty`: Number of product photos published
    - `product_weight_g`: Product weight in grams
    - `product_length_cm`: Product length in centimeters
    - `product_height_cm`: Product height in centimeters
    - `product_width_cm`: Product width in centimeters

### Data Quality

- There are **no duplicate rows** in this dataset.
- Missing values are found in the following columns:
    - `product_category_name`
    - `product_name_lenght`
    - `product_description_lenght`
    - `product_photos_qty`
    - Several product dimension attributes (`product_weight_g`, `product_length_cm`, `product_height_cm`, `product_width_cm`)

### Insights

- This dataset provides comprehensive details about products, including dimensions, weight, and categories. 
- The presence of missing values in certain attributes might affect analysis related to product specifics.

### Next Steps

- Address missing values by imputation or exclusion, depending on the analysis requirements.
- Investigate the impact of missing values on product-related analyses, such as weight or dimension-based evaluations.
- Consider merging this dataset with others to enrich product-related insights.

## Sellers dataset

In [37]:
# Sellers dataset
sellers_df = pd.read_csv('/Users/irasafonik/SMART_business/sellers.csv')
sellers_df.head()

seller_id  seller_zip_code_prefix  \
0  3442f8959a84dea7ee197c632cb2df15                   13023   
1  d1b65fc7debc3361ea86b5f14c68d2e2                   13844   
2  ce3ad9de960102d0677a81f5d0bb7b2d                   20031   
3  c0f3eea2e14555b6faeea3dd58c1b1c3                    4195   
4  51a04a8a6bdcb23deccc82b0b80742cf                   12914   

         seller_city seller_state  
0           campinas           SP  
1         mogi guacu           SP  
2     rio de janeiro           RJ  
3          sao paulo           SP  
4  braganca paulista           SP

In [38]:
sellers_df.info()
sellers_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3095 entries, 0 to 3094
Data columns (total 4 columns):
 #   Column                  Non-Null Count  Dtype 
---  ------                  --------------  ----- 
 0   seller_id               3095 non-null   object
 1   seller_zip_code_prefix  3095 non-null   int64 
 2   seller_city             3095 non-null   object
 3   seller_state            3095 non-null   object
dtypes: int64(1), object(3)
memory usage: 96.8+ KB


seller_id                 0
seller_zip_code_prefix    0
seller_city               0
seller_state              0
dtype: int64

In [74]:
sellers_df.duplicated().sum()

0

## 📝 Conclusion

### Sellers Dataset Overview

- The dataset contains **3,095 entries** with **4 columns**:
    - `seller_id`: Unique identifier for the seller
    - `seller_zip_code_prefix`: First 5 digits of the seller's zip code
    - `seller_city`: City where the seller is located
    - `seller_state`: State where the seller is located

### Data Quality

- There are **no missing values** in this dataset.
- There are **no duplicates** in this dataset.

### Insights

- This dataset provides detailed information on seller locations, which can be used for various analyses, such as regional performance and logistics planning.

### Next Steps

- Integrate this dataset with order data to analyze seller performance by location.
- Map seller locations to visualize distribution and identify regional patterns.

## Product category name translation dataset

In [40]:
# Product category name translation dataset
category_translation_df = pd.read_csv('/Users/irasafonik/SMART_business/product_category_name_translation.csv')
category_translation_df.head()

product_category_name product_category_name_english
0            beleza_saude                 health_beauty
1  informatica_acessorios         computers_accessories
2              automotivo                          auto
3         cama_mesa_banho                bed_bath_table
4        moveis_decoracao               furniture_decor

In [41]:
category_translation_df.info()
category_translation_df.isnull().sum()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 71 entries, 0 to 70
Data columns (total 2 columns):
 #   Column                         Non-Null Count  Dtype 
---  ------                         --------------  ----- 
 0   product_category_name          71 non-null     object
 1   product_category_name_english  71 non-null     object
dtypes: object(2)
memory usage: 1.2+ KB


product_category_name            0
product_category_name_english    0
dtype: int64

In [75]:
category_translation_df.duplicated().sum()

0

## 📝 Conclusion

### Product Categories Dataset Overview

- The dataset contains **72 entries** with **2 columns**:
    - `product_category_name`: Name of the product category in the original language
    - `product_category_name_english`: Name of the product category in English

### Data Quality

- There are **no missing values** in this dataset.
- There are **no duplicate rows** in this dataset.

### Insights

- This dataset provides a comprehensive list of product categories in both the original and English languages.
- It allows for easier integration and analysis of product categories across different datasets.

### Next Steps

- Use this dataset to standardize product category names for consistency across various analyses.
- Merge this dataset with other product-related data to enrich category information and perform more detailed analysis.